src/query/storage/m3/cluster_resolver.go

```diff
@@ -463,3 +500,19 @@ func newCoversRangeFilter(opts coversRangeFilterOptions) func(namespace ClusterN
 		return !clusterStart.After(opts.queryStart)
 	}
 }
+
+type resolvedNamespacesByResolutionAsc []resolvedNamespace
```

Comment:
nit: we could define a type `type resolvedNamespaces []resolvedNamespace` and implement these methods (add a test if not exists to ensure it's in asc order`. We can then add a constructor to construct resolvedNamespaces instead of []resolveNamespace[]{resolved(clusterNamespace)}`. I think it would be cleaner.

Reply:
There is both `resolvedNamespacesByResolutionAsc` and `resolvedNamespacesByRetentionAsc`, need to differentiate.